app/controllers/application_controller.rb

```diff
@@ -0,0 +1,12 @@
+class ApplicationController < ActionController::API
+  # include ActionController::Serialization
+  before_action :get_user
+
+  def get_user
+    if User.exists?(username: request.headers[:username])
```

Comment:
this query is redundant. WE can simply the user if the header value is present and return the error is the find returned no record, instead of firing 2 queries.

Reply:
Will change if condition to find_by